File: pr/2laboratory7c++/1.cpp
#include <iostream>

class Rectangle {
private:
    double length;
    double width;

public:
    // Конструктор по умолчанию
    Rectangle() : length(1), width(1) {}

    // Параметризованный конструктор
    Rectangle(double len, double wdt) : length(len), width(wdt) {}

    // Метод для вычисления площади
    double area() const {
        return length * width;
    }

    // Метод для вычисления периметра
    double perimeter() const {
        return 2 * (length + width);
    }
};

int main() {
    // Создание объектов класса Rectangle
    Rectangle rect1; // Использование конструктора по умолчанию
    Rectangle rect2(5, 3); // Использование параметризованного конструктора
    Rectangle rect3(4.5, 2.1);

    // Вывод площади и периметра для каждого прямоугольника
    std::cout << "Rectangle 1 - Area: " << rect1.area() << ", Perimeter: " << rect1.perimeter() << std::endl;
    std::cout << "Rectangle 2 - Area: " << rect2.area() << ", Perimeter: " << rect2.perimeter() << std::endl;
    std::cout << "Rectangle 3 - Area: " << rect3.area() << ", Perimeter: " << rect3.perimeter() << std::endl;

    return 0;
}

File: pr/2aaboratory2go/1.go
package main

import (
	"fmt"
)

type rectangle struct {
	width float32
	height float32
}

func (_rectangle *rectangle) area() (float32) {
	return _rectangle.width * _rectangle.height
}

func (_rectangle *rectangle) perimeter() (float32) {
	return (_rectangle.width + _rectangle.height) * 2
}

func main() {
	bigRectangle := rectangle{100,10}
	fmt.Println(bigRectangle.area())
	fmt.Println(bigRectangle.perimeter())
}